### DISCUSSION TITLE
Switch from default #[tokio::main] to single-threaded

### DISCUSSION THREAD

**Discussion by ScottLinnn** [tokio-rs/tokio#4718] [Q&A] [answered]
I have several microservices using the default `#[tokio::main]`, and I want to change all of them to be single-threaded. While I can manually change the annotations to `#[tokio::main(flavor = "current_thread")]`, I'm wondering is there a way to change this configuration without changing the annotations themselves? For example, I'm using `axum`, `tower `to set up middleware, is it possible to change runtime configuration there?

Thank you!

**Comment by Darksonn** [comment, marked as answer]
No. The only way to change this option is to set the `flavor` on the annotation itself.

In the past this was actually possible by disabling the `rt-multi-thread` feature of the Tokio crate, but this was changed because it caused a lot of confusion.

**Reply by ScottLinnn** [reply]
Thank you!
